l1-contracts/src/core/Rollup.sol

```diff
@@ -35,6 +35,16 @@ import {Leonidas} from "./sequencer_selection/Leonidas.sol";
 contract Rollup is Leonidas, IRollup, ITestRollup {
   using SafeCast for uint256;
 
+  struct ChainTip {
+    uint256 blockNumber;
+    uint256 slotNumber;
```

Comment:
Why do we need the slot numbers here? 

If you have the block number, you can always go and look it up from there, so this would just add extra cost on storage.